api/app/registrationResponse/repository.py

```diff
@@ -45,3 +49,18 @@ def get_confirmed_for_event(event_id, confirmed):
         ).filter(
             Offer.event_id == event_id
         ).all()
+
+    @staticmethod
+    def get_confirmed_for_event_exclude_completed_registrations(event_id, confirmed):
+        """Get registrations for an event according to confirmed status, excluding those who have already registered."""
```

Comment:
This is a little confusing because here I think you're using registered to mean "showed up at the event", but elsewhere registered just means "have confirmed they're coming and agreed to the code of conduct and filled in some details"

Reply:
I have changed the name and made it more descriptive. This just gets people who have confirmed they are attending, but not signed in yet.